File: testdata/jennies/builders/foreign_builder/PHPBuilder/src/BuilderPkg/SomeNiceBuilderBuilder.php
<?php

namespace Grafana\Foundation\BuilderPkg;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\SomePkg\SomeStruct>
 */
class SomeNiceBuilderBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\SomePkg\SomeStruct $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\SomePkg\SomeStruct();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\SomePkg\SomeStruct
     */
    public function build()
    {
        return $this->internal;
    }

    public function title(string $title): static
    {
        $this->internal->title = $title;
    
        return $this;
    }

}

File: testdata/jennies/builders/foreign_builder/JavaBuilder/builder_pkg/SomeNiceBuilderBuilder.java
package builder_pkg;

import some_pkg.SomeStruct;

public class SomeNiceBuilderBuilder implements cog.Builder<SomeStruct> {
    protected final SomeStruct internal;
    
    public SomeNiceBuilderBuilder() {
        this.internal = new SomeStruct();
    }
    public SomeNiceBuilderBuilder title(String title) {
        this.internal.title = title;
        return this;
    }
    public SomeStruct build() {
        return this.internal;
    }
}
